lib/ansible/modules/system/macos_pref.py

```diff
@@ -0,0 +1,372 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018 - Matthias Hollerbach <mail@matthias-hollerbach.de>
+# Copyright (c) 2016 - John Calixto <john.calixto@nordstrom.com> and Nordstrom, Inc.
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: macos_pref
+author:
+    - Matthias Hollerbach (@kinglouie)
+    - John Calixto (@nordjc)
+short_description: Manipulates macOS preferences including complex data types.
+version_added: "2.6"
+description:
+    - This module allows users to create, read, update, and delete system and application preferences on macOS
+      including deeply nested dictionary values like those in the C(com.apple.finder) domain.
+    - It also provides a convenient C(merge) strategy for assignments in nested dictionaries so that users can very
+      specifically target nested keys without having to assign unchanged peer or parent values.
+requirements:
+    - Target machine should be running macOS
```

Comment:
Any limitations on what is the minimal OS version supported, or should be work with pretty much anything?

Reply:
I couldn't find any info what the minimum version of macOS/OS X is where PyObjC is supported, I will investigate that matter.

Reply:
PyObjC was included in Mac OS X from version 10.5 (Leopard), but python included in 10.5 was version 2.5. Personally, I've developed my `osx_defaults` version #24808 on Mac OS X 10.6 (python 2.6) but test it and use it also on macOS 10.11, 10.12 and 10.13 (python 2.7).

Reply:
Well, I guess that anything that's actually still supported will work. Maybe mentioning minimal requirement as 10.11+?